### PR TITLE
Widen MonoJumpInfo union to pointer-sized so mono_add_patch_info can just use target field blindly.

### PR DESCRIPTION
**Pull request by jaykrell**
Alternative to https://github.com/mono/mono/pull/14641.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by NattyNarwhal** [comment]
Also seems to work for me.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Windows x64

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Windows i386

File: mono/metadata/jit-icall-reg.h [hunk 1]
```diff
@@ -339,7 +339,13 @@ MONO_JIT_ICALL (ves_icall_thread_finish_async_abort) \
 	\
 MONO_JIT_ICALL (count) \
 
-typedef enum MonoJitICallId {
+// Widen to gsize for use in MonoJumpInfo union.
+#ifdef __cplusplusplus
```
Comment (filipnavara) [inline mono/metadata/jit-icall-reg.h]: Is the C+++ intentional?

Comment (jaykrell) [reply mono/metadata/jit-icall-reg.h]: Yes.

Comment (jaykrell) [reply mono/metadata/jit-icall-reg.h]: Oops just C++, not C+++, fixed, thank you.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Windows fails for lack of cmake to build BTLS.

**Pull request merged**